tests-arquillian/src/test/java/org/jboss/weld/tests/contexts/propagation/ContextPropagationTest.java

```diff
@@ -104,4 +111,22 @@ private void pingBeanAndOffloadTask(WeldManager manager, Class<? extends Abstrac
             Assert.fail(e.toString());
         }
     }
+
+    private void pingBeanAndRunImmediately(WeldManager manager, Class<? extends AbstractBeanWithState> beanClazz) {
+        // use the bean in this thread first - set counter to two
+        AbstractBeanWithState bean = manager.instance().select(beanClazz).get();
+        Assert.assertEquals(0, bean.getValue());
+        bean.incrementCounter();
+        bean.incrementCounter();
+        Assert.assertEquals(2, bean.getValue());
+
+        // prepare a callable which will further increase the counter and return the value we found there
+        Callable<Integer> callableTask = () -> {
+            AbstractBeanWithState beanInCallable = CDI.current().select(beanClazz).get();
```

Comment:
I think you can safely replace `CDI.current().select(beanClazz).get()` with `bean`... but your version is perfectly ok too ;-)

Reply:
Yeah I could, that's really just a copy of the `pingBeanAndOffloadTask()` method...